From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sat, 08 Aug 2016 15:06:00 +0200
Subject: [PATCH] Fix linux_musl_base for native musl host

See https://github.com/rust-lang/rust/pull/40113

diff -Naur rustc-1.75.0-src.orig/compiler/rustc_target/src/spec/base/linux_musl.rs rustc-1.75.0-src/compiler/rustc_target/src/spec/base/linux_musl.rs
--- rustc-1.75.0-src.orig/compiler/rustc_target/src/spec/base/linux_musl.rs	2024-01-19 10:27:17.735163633 +0100
+++ rustc-1.75.0-src/compiler/rustc_target/src/spec/base/linux_musl.rs	2024-01-19 10:28:30.779887528 +0100
@@ -1,15 +1,12 @@
-use crate::spec::{base, crt_objects, LinkSelfContainedDefault, TargetOptions};
+use crate::spec::{base, TargetOptions};
 
 pub fn opts() -> TargetOptions {
     let mut base = base::linux::opts();
 
     base.env = "musl".into();
-    base.pre_link_objects_self_contained = crt_objects::pre_musl_self_contained();
-    base.post_link_objects_self_contained = crt_objects::post_musl_self_contained();
-    base.link_self_contained = LinkSelfContainedDefault::InferredForMusl;
 
     // These targets statically link libc by default
-    base.crt_static_default = true;
+    base.crt_static_default = false;
 
     base
 }
diff -Naur rustc-1.75.0-src.orig/compiler/rustc_target/src/spec/crt_objects.rs rustc-1.75.0-src/compiler/rustc_target/src/spec/crt_objects.rs
--- rustc-1.75.0-src.orig/compiler/rustc_target/src/spec/crt_objects.rs	2024-01-19 10:27:17.746163591 +0100
+++ rustc-1.75.0-src/compiler/rustc_target/src/spec/crt_objects.rs	2024-01-19 10:27:30.341115881 +0100
@@ -61,28 +61,6 @@
     ])
 }
 
-pub(super) fn pre_musl_self_contained() -> CrtObjects {
-    new(&[
-        (LinkOutputKind::DynamicNoPicExe, &["crt1.o", "crti.o", "crtbegin.o"]),
-        (LinkOutputKind::DynamicPicExe, &["Scrt1.o", "crti.o", "crtbeginS.o"]),
-        (LinkOutputKind::StaticNoPicExe, &["crt1.o", "crti.o", "crtbegin.o"]),
-        (LinkOutputKind::StaticPicExe, &["rcrt1.o", "crti.o", "crtbeginS.o"]),
-        (LinkOutputKind::DynamicDylib, &["crti.o", "crtbeginS.o"]),
-        (LinkOutputKind::StaticDylib, &["crti.o", "crtbeginS.o"]),
-    ])
-}
-
-pub(super) fn post_musl_self_contained() -> CrtObjects {
-    new(&[
-        (LinkOutputKind::DynamicNoPicExe, &["crtend.o", "crtn.o"]),
-        (LinkOutputKind::DynamicPicExe, &["crtendS.o", "crtn.o"]),
-        (LinkOutputKind::StaticNoPicExe, &["crtend.o", "crtn.o"]),
-        (LinkOutputKind::StaticPicExe, &["crtendS.o", "crtn.o"]),
-        (LinkOutputKind::DynamicDylib, &["crtendS.o", "crtn.o"]),
-        (LinkOutputKind::StaticDylib, &["crtendS.o", "crtn.o"]),
-    ])
-}
-
 pub(super) fn pre_mingw_self_contained() -> CrtObjects {
     new(&[
         (LinkOutputKind::DynamicNoPicExe, &["crt2.o", "rsbegin.o"]),
